core/statistics/tpsBenchmark.go

```diff
@@ -179,8 +179,13 @@ func (s *TpsBenchmark) NrOfShards() uint32 {
 // ShardStatistics returns the current statistical state for a given shard
 func (s *TpsBenchmark) ShardStatistics() map[uint32]ShardStatistic {
 	s.mut.RLock()
-	defer s.mut.RUnlock()
-	return s.shardStatistics
+	copyMapShardStatistics := make(map[uint32]ShardStatistic)
```

Comment:
Seems fine, I see that `updateStatistics` is protected by the same lock.

Reply:
rename to `shardStatisticsMapCopy`

Reply:
done.